src/dune_pkg/rev_store.ml

```diff
@@ -33,7 +33,13 @@ let with_flock lock_path ~f =
   let open Fiber.O in
   let parent = Path.parent_exn lock_path in
   Path.mkdir_p parent;
-  let fd = Unix.openfile (Path.to_string lock_path) [ Unix.O_CREAT; O_RDONLY ] 0o644 in
+  let fd =
+    Unix.openfile
+      (Path.to_string lock_path)
+      [ Unix.O_CREAT; O_RDONLY; O_SHARE_DELETE ]
+      0o600
+  in
+  Unix.set_close_on_exec fd;
```

Comment:
I think `O_CLOEXEC` would be better in that case, since this also avoids potential race conditions.

Reply:
ah yes indeed.